packages/react-server/src/ReactFizzServer.js

```diff
@@ -2063,20 +2069,34 @@ function flushCompletedQueues(
 
     // TODO: Emit preloading.
 
-    // TODO: It's kind of unfortunate to keep checking this array after we've already
-    // emitted the root.
+    let i;
     const completedRootSegment = request.completedRootSegment;
-    if (completedRootSegment !== null && request.pendingRootTasks === 0) {
-      flushSegment(request, destination, completedRootSegment);
-      request.completedRootSegment = null;
-      writeCompletedRoot(destination, request.responseState);
+    if (completedRootSegment !== null) {
+      if (request.pendingRootTasks === 0) {
+        if (enableFloat) {
+          const preamble: Array<
+            Chunk | PrecomputedChunk,
+          > = (request.preamble: any);
+          for (i = 0; i < preamble.length; i++) {
+            // we expect the preamble to be tiny and will ignore backpressure
+            writeChunk(destination, preamble[i]);
+          }
+          preamble.length = 0;
```

Comment:
Why bother resetting this? The memory will still be retained.

Reply:
I think a holdover from when this wasn't gated in a flush once block but agree it can be removed